colcon_ros/task/catkin/build.py

```diff
@@ -98,12 +99,20 @@ def add_arguments(self, *, parser):  # noqa: D102
                 try:
                     # try to set CATKIN_ENV_HOOK_WORKSPACE explicitly before
                     # sourcing these hooks
-                    additional_hooks.append(
-                        shell_extension.create_hook_set_value(
-                            'catkin_env_hook_workspace',
-                            Path(args.install_base), self.context.pkg.name,
-                            'CATKIN_ENV_HOOK_WORKSPACE',
-                            '$COLCON_CURRENT_PREFIX'))
+                    if sys.platform == 'win32':
+                        additional_hooks.append(
+                            shell_extension.create_hook_set_value(
+                                'catkin_env_hook_workspace',
+                                Path(args.install_base), self.context.pkg.name,
+                                'CATKIN_ENV_HOOK_WORKSPACE',
+                                '%COLCON_CURRENT_PREFIX%'))
+                    else:
+                        additional_hooks.append(
+                            shell_extension.create_hook_set_value(
+                                'catkin_env_hook_workspace',
+                                Path(args.install_base), self.context.pkg.name,
+                                'CATKIN_ENV_HOOK_WORKSPACE',
+                                '$COLCON_CURRENT_PREFIX'))
```

Comment:
Please avoid duplicating multiple lines of code. Instead only conditionally determine the value and then use a single code block to call the function with that value.